sites/public/lib/helpers.tsx

```diff
@@ -64,6 +65,45 @@ export const getListings = (listings) => {
     rent: t("t.rent"),
     availability: t("t.availability"),
   }
+
+  const { profile, userProfileService } = useContext(AuthContext)
+
+  const addToFavorite = async (listingID: string) => {
+    const preferences: UserPreferences = profile?.preferences || { favorites: [] }
+
+    if (!preferences.favorites?.includes(listingID)) {
+      preferences.favorites?.push(listingID)
+    }
+    console.log(preferences?.favorites)
+
+    try {
+      await userProfileService?.update({
+        body: { ...profile, preferences },
+      })
+    } catch (err) {
+      console.warn(err)
+    }
+  }
+
+  const removeFavorite = async (listingID: string) => {
```

Comment:
Similarly here, I think this can be simplified to just pass the ID to be removed.